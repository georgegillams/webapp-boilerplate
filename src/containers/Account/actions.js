import { createRoutine } from 'redux-saga-routines';

import { LOAD_SHOWCASES } from './constants';

export const loadShowcases = createRoutine(LOAD_SHOWCASES);
