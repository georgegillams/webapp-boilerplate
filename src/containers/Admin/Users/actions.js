import { createRoutine } from 'redux-saga-routines';

import { LOAD, REMOVE, REQUEST_MAGIC_LINK } from './constants';

export const load = createRoutine(LOAD);
export const remove = createRoutine(REMOVE);
export const requestMagicLink = createRoutine(REQUEST_MAGIC_LINK);
