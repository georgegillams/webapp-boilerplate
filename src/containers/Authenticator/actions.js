import { createRoutine } from 'redux-saga-routines';

import { LOAD_AUTH, SET_COOKIES_ALLOWED } from './constants';

export const loadAuth = createRoutine(LOAD_AUTH);
export const setCookiesAllowed = createRoutine(SET_COOKIES_ALLOWED);
