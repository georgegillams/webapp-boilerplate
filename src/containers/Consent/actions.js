import { createRoutine } from 'redux-saga-routines';

import { CONSENT, DEFER_CONSENT, RE_PROMPT_CONSENT, RESET_CONSENT } from './constants';

export const consent = createRoutine(CONSENT);
export const deferConsent = createRoutine(DEFER_CONSENT);
export const rePromptConsent = createRoutine(RE_PROMPT_CONSENT);
export const resetConsent = createRoutine(RESET_CONSENT);
