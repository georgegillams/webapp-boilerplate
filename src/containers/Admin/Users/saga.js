import { takeLatest, put, call, select } from 'redux-saga/effects';

import request from 'utils/request';
import apiStructure from 'helpers/apiStructure';

import { selectState } from './selectors';
import { load, remove, requestMagicLink } from './actions';

export function* doLoad() {
  const requestURL = apiStructure.loadUsers.fullPath;

  try {
    yield put(load.request());

    const result = yield call(request, requestURL);

    if (result.error) {
      yield put(load.failure(result));
    } else {
      yield put(load.success(result.users));
    }
  } catch (err) {
    yield put(load.failure(err));
  } finally {
    yield put(load.fulfill());
  }
}

export function* doRemove() {
  const currentState = yield select(selectState());
  const { userToRemove } = currentState;
  const requestURL = apiStructure.deleteUser.fullPath;

  try {
    yield put(remove.request());

    const result = yield call(request, requestURL, {
      method: 'POST',
      body: JSON.stringify(userToRemove),
      headers: {
        'Content-Type': 'application/json',
      },
    });

    if (result.error) {
      yield put(remove.failure(result));
    } else {
      yield put(remove.success(result));
      yield put(load.trigger());
    }
  } catch (err) {
    yield put(remove.failure(err));
  } finally {
    yield put(remove.fulfill());
  }
}

export function* doRequestMagicLink() {
  const currentState = yield select(selectState());
  const { userToLogIn } = currentState;
  const requestURL = apiStructure.requestMagicLink.fullPath;

  try {
    yield put(requestMagicLink.request());

    const result = yield call(request, requestURL, {
      method: 'POST',
      body: JSON.stringify({ email: userToLogIn.email, divertToAdmin: true }),
      headers: {
        'Content-Type': 'application/json',
      },
    });

    if (result.error) {
      yield put(requestMagicLink.failure(result));
    } else {
      yield put(requestMagicLink.success(result));
    }
  } catch (err) {
    yield put(requestMagicLink.failure(err));
  } finally {
    yield put(requestMagicLink.fulfill());
  }
}

export default function* saga() {
  yield takeLatest(load.TRIGGER, doLoad);
  yield takeLatest(remove.TRIGGER, doRemove);
  yield takeLatest(requestMagicLink.TRIGGER, doRequestMagicLink);
}
