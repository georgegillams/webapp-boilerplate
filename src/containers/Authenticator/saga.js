import { takeLatest, put, call } from 'redux-saga/effects';

import request from 'utils/request';
import apiStructure from 'helpers/apiStructure';

import { loadAuth } from './actions';

export function* doLoadAuth() {
  // const currentState = yield select(selectState());
  // console.log(`currentState`, currentState);

  const requestURL = apiStructure.loadAuth.fullPath;

  try {
    yield put(loadAuth.request());

    const result = yield call(request, requestURL);

    yield put(loadAuth.success(result));
  } catch (err) {
    yield put(loadAuth.failure(err));
  } finally {
    yield put(loadAuth.fulfill());
  }
}

export default function* saga() {
  yield takeLatest(loadAuth.TRIGGER, doLoadAuth);
}
