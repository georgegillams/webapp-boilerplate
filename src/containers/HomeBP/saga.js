import { takeLatest, put, call } from 'redux-saga/effects';

import request from 'utils/request';

import { loadShowcases } from './actions';

export function* loadShowcasesRequest() {
  const requestURL = 'https://us-central1-react-next-boilerplate-cda8b.cloudfunctions.net/getShowcasesData';

  try {
    yield put(loadShowcases.request());

    const result = yield call(request, requestURL);

    yield put(loadShowcases.success(result));
  } catch (err) {
    yield put(loadShowcases.failure(err));
  } finally {
    yield put(loadShowcases.fulfill());
  }
}

export default function* saga() {
  yield takeLatest(loadShowcases.TRIGGER, loadShowcasesRequest);
}
