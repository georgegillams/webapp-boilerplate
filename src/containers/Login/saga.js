import { takeLatest, put, call, select } from 'redux-saga/effects';

import request from 'utils/request';
import apiStructure from 'helpers/apiStructure';
import { selectState } from './selectors';

import { login } from './actions';

export function* doLogin() {
  const currentState = yield select(selectState());
  const { credentials } = currentState;

  const requestURL = apiStructure.requestMagicLink.fullPath;

  try {
    yield put(login.request());

    // TODO should pass loginRedirect to the API
    const result = yield call(request, requestURL, {
      method: 'POST',
      body: JSON.stringify({ ...credentials }),
      headers: {
        'Content-Type': 'application/json',
      },
    });

    yield put(login.success(result));
  } catch (err) {
    yield put(login.failure(err));
  } finally {
    yield put(login.fulfill());
  }
}

export default function* saga() {
  yield takeLatest(login.TRIGGER, doLogin);
}
