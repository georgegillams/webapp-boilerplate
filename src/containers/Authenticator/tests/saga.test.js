import { put, takeLatest } from 'redux-saga/effects';

import { loadAuth } from '../actions';

import saga, { doLoadAuth } from '../saga';

describe('loadAuthRequest Saga', () => {
  let mainSaga;

  beforeEach(() => {
    mainSaga = saga();
  });

  it('Should get authenticated user TRIGGER', () => {
    const takeLatestDescriptor = mainSaga.next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest(loadAuth.TRIGGER, doLoadAuth));
  });

  describe('loadAuthenticator actions', () => {
    let loadAuthGenerate;

    const response = {
      name: 'userName',
      status: 200,
    };

    beforeEach(() => {
      loadAuthGenerate = doLoadAuth();

      const selectDescriptor = loadAuthGenerate.next().value;
      expect(selectDescriptor).toMatchSnapshot();
    });

    it('Should loadAuthRequest be success', () => {
      loadAuthGenerate.next(response);
      const putSuccess = loadAuthGenerate.next(response).value;
      loadAuthGenerate.next(response);

      expect(putSuccess).toEqual(put(loadAuth.success(response)));
    });

    it('Should loadAuthRequest be failure', () => {
      const response = new Error('Some error');
      const putFailure = loadAuthGenerate.throw(response).value;

      expect(putFailure).toEqual(put(loadAuth.failure(response)));
    });
  });
});
