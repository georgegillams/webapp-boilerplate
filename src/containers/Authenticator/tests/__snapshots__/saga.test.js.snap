// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadAuthRequest Saga loadAuthenticator actions Should loadAuthRequest be failure 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "action": Object {
      "type": "app/authenticator/LOAD_AUTH/REQUEST",
    },
    "channel": undefined,
  },
  "type": "PUT",
}
`;

exports[`loadAuthRequest Saga loadAuthenticator actions Should loadAuthRequest be success 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "action": Object {
      "type": "app/authenticator/LOAD_AUTH/REQUEST",
    },
    "channel": undefined,
  },
  "type": "PUT",
}
`;
